src/components/PopupWithForm.js

```diff
@@ -5,18 +5,23 @@ export class PopupWithForm extends Popup {
     this._popUpForm = this._popUpElement.querySelector(".modal__form");
     this._handleModalSubmit = handleModalSubmit;
   }
-  closeModal() {
-    this._popUpForm.reset();
-    super.closeModal();
-  }
+  // closeModal() {
+  //   this._popUpForm.reset();
+  //   super.closeModal();
+  // }
   setEventListeners() {
     super.setEventListeners();
     this._popUpForm.addEventListener("submit", (e) => {
-      e.preventDefault();
       const modalInputs = this._getInputValues();
-      this._handleModalSubmit(modalInputs);
+      e.preventDefault();
+      this._popUpForm.querySelector(".modal__form-button").innerText =
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```